tapgarden/batch.go

```diff
@@ -309,6 +315,35 @@ func (m *MintingBatch) HasSeedlings() bool {
 	return len(m.Seedlings) != 0
 }
 
+// ValidateSeedling checks if a seedling is valid for the batch.
+func (m *MintingBatch) ValidateSeedling(newSeedling Seedling) error {
+	// Ensure that the seedling and batch agree on the enabled universe
+	// announcements.
+	if m.EnableUniAnnounce != newSeedling.EnableUniAnnounce {
+		return fmt.Errorf("batch and seedling do not agree on " +
+			"enabled universe announcements")
+	}
+
+	// If the seedling supports universe announcements, it must have a group
+	// anchor or the same group key as all the other seedlings in the batch.
+	if newSeedling.EnableUniAnnounce {
```

Comment:
👌🏽 

This constraint could be loosened in the future but it seems reasonable for this initial version, assuming that it doesn't affect the proof format or on-chain TX pattern.